google-apis-core/lib/google/apis/core/api_command.rb

```diff
@@ -172,6 +174,10 @@ def set_user_project_header
           header['X-Goog-User-Project'] = quota_project_id if quota_project_id
         end
 
+        def invocation_id_header
```

Comment:
Will this header be consistent across retry attempts? I don't totally see how that would happen based on the code here. Is it based on `add_invocation_id_header` being called outside the retry loop in the manual client?

It would be good to add docs to clarify this, and a test that the value is consistent across retry attempts (in the veneer client if necessary).

Reply:
Thanks for clarifying and adding the test case!

As far as docs, I see that you actually have a comment in options.rb below which seems like plenty.